include/IntervalNode.hpp

```diff
@@ -143,23 +151,27 @@ void construct_tree(std::unique_ptr<IntervalNode> & node, std::vector<Record> co
 
     // Calculate and set median.
     node->set_median(calculate_median(records_i));
+    Position cur_median = node->get_median();
```

Comment:
I thought so too, but set_median will `std::move` the parameter. So I think we can't call set_median on the cur_median, because then the contents of cur_median will be moved.

Reply:
Ok, but maybe we should rethink the set function then at some point.